oak/server/rust/oak_runtime/src/config.rs

```diff
@@ -14,121 +14,38 @@
-    module_name_wasm: HashMap<String, Vec<u8>, S>,
-    logger_name: &str,
-    initial_node: &str,
-    entrypoint: &str,
+pub fn application_configuration(
+    wasm_modules: HashMap<String, Vec<u8>>,
+    initial_node_configuration: &NodeConfiguration,
 ) -> ApplicationConfiguration {
-    let mut nodes: Vec<NodeConfiguration> = module_name_wasm
-        .into_iter()
-        .sorted()
-        .map(|(name, wasm)| NodeConfiguration {
-            name,
-            config_type: Some(ConfigType::WasmConfig(WebAssemblyConfiguration {
-                module_bytes: wasm,
-            })),
-        })
-        .collect();
-
-    if !logger_name.is_empty() {
-        nodes.push(NodeConfiguration {
-            name: logger_name.to_string(),
-            config_type: Some(ConfigType::LogConfig(LogConfiguration {})),
-        });
-    }
-
     ApplicationConfiguration {
-        node_configs: nodes,
-        initial_node_config_name: initial_node.into(),
-        initial_entrypoint_name: entrypoint.into(),
-        ..Default::default()
+        wasm_modules,
+        initial_node_configuration: Some(initial_node_configuration.clone()),
+        grpc_port: 0,
     }
 }
 
-/// Load a `runtime::Configuration` from a protobuf `ApplicationConfiguration`.
-/// This can fail if an unsupported Node is passed, or if a Node was unable to be initialized with
-/// the given configuration.
-pub fn from_protobuf(
-    app_config: ApplicationConfiguration,
-) -> Result<runtime::Configuration, OakStatus> {
-    let mut config = runtime::Configuration {
-        nodes: HashMap::new(),
-        entry_module: app_config.initial_node_config_name.clone(),
-        entrypoint: app_config.initial_entrypoint_name.clone(),
-    };
-
-    for node in app_config.node_configs {
-        config.nodes.insert(
-            node.name.clone(),
-            match &node.config_type {
-                None => {
-                    error!("Node config {} with no type", node.name);
-                    return Err(OakStatus::ErrInvalidArgs);
-                }
-                Some(ConfigType::LogConfig(_)) => node::Configuration::LogNode,
-                Some(ConfigType::GrpcServerConfig(GrpcServerConfiguration { address })) => address
-                    .parse()
-                    .map_err(|error: AddrParseError| error.into())
-                    .and_then(|address| check_port(&address).map(|_| address))
-                    .map(|address| node::Configuration::GrpcServerNode { address })
-                    .map_err(|error| {
-                        error!("Incorrect gRPC server address: {:?}", error);
-                        OakStatus::ErrInvalidArgs
-                    })?,
-                Some(ConfigType::WasmConfig(WebAssemblyConfiguration { module_bytes, .. })) => {
-                    load_wasm(&module_bytes).map_err(|e| {
-                        error!("Error loading Wasm module: {}", e);
-                        OakStatus::ErrInvalidArgs
```

Comment:
I like not having the data structures that mirror the protobuf config structure (and I think this would fix #755 along the way); are the validity checks (`check_port`, `load_wasm`) now done at a later point?

[Edit: yes they are]

Reply:
Affirmative.